### DISCUSSION TITLE
Bridging Twitch Chat (based on IRC) into a Matrix or XMPP Room

### DISCUSSION THREAD

**Discussion by linux-lukas** [42wim/matterbridge#1637] [General]
It came to my attention through an acquaintance of mine that the Twitch chat is based on IRC.

The aforementioned acquaintance drew my attention to this project when I was talking to him about it.

EDIT: I saw that Twitch is probably supported - to what extent?

References: https://dev.twitch.tv/docs/irc

**Comment by Mikaela** [comment]
Yes, this is possible and I am using matterbridge to connect Twitch to both Matrix and XMPP. It's also documented in the wiki.

* https://github.com/42wim/matterbridge/wiki/Section-Twitch-(basic)

**Reply by linux-lukas** [reply]
I have a question of understanding: So: the person streaming sets up for a room incl. Matterbridge - so only they need an account and then the whole chat is bridged into this room, right @Mikaela?

If now Matrix users (i.e. without a Twitch account) come into the room and write - will this also be displayed in the Twitch chat?

**Reply by Mikaela** [reply]
Matterbridge needs it's own account, but otherwise yes, it works just like IRC as documented in the wiki.

* https://mikaela.info/discuss has my setup

**Comment by linux-lukas** [comment]
How does bridging messages from Matrix/Element users to Twitch work WITHOUT a Twitch account?

**Reply by Mikaela** [reply]
The bot/matterbridge must have an Twitch account. Twitch users see messages coming from it in the configured format, likely beginning with RemoteNickFormat

**Reply by linux-lukas** [reply]
Okay, sounds very exciting. xD
